src/datasources/asset-calibration/AssetCalibrationDataSource.ts

```diff
@@ -50,35 +50,70 @@ export class AssetCalibrationDataSource extends DataSourceBase<AssetCalibrationQ
     super(instanceSettings, backendSrv, templateSrv);
   }
 
+  async runQuery(query: AssetCalibrationQuery, options: DataQueryRequest): Promise<DataFrameDTO> {
+    await this.loadDependencies();
+
+    if (query.filter) {
+      query.filter = transformComputedFieldsQuery(
+        this.templateSrv.replace(query.filter, options.scopedVars),
+        this.assetComputedDataFields,
+        this.queryTransformationOptions
+      );
+    }
+
+    return await this.processCalibrationForecastQuery(query as AssetCalibrationQuery, options);
+  }
+
   private readonly assetComputedDataFields = new Map<AssetCalibrationFieldNames, ExpressionTransformFunction>([
+    ...Object.values(AssetCalibrationFieldNames).map(field => [field, this.multipleValuesQuery(field)] as [AssetCalibrationFieldNames, ExpressionTransformFunction]),
     [
       AssetCalibrationFieldNames.LOCATION,
       (value: string, operation: string, options?: TTLCache<string, unknown>) => {
+        let values = [value];
+
+        if (value.includes('{') && value.includes('}')) {
+          values = this.getMultipleValuesArray(value);
+        }
+
+        if (values.length > 1) {
+          return `(${values.map(val => `Location.MinionId ${operation} "${val}"`).join(` ${this.getLocicalOperator(operation)} `)})`;
+        }
+
         if (options?.has(value)) {
           return `Location.MinionId ${operation} "${value}"`
         }
 
-        const logicalOperator = operation === QueryBuilderOperations.EQUALS.name ? '||' : '&&';
-        return `(Location.MinionId ${operation} "${value}" ${logicalOperator} Location.PhysicalLocation ${operation} "${value}")`;
+        return `(Location.MinionId ${operation} "${value}" ${this.getLocicalOperator(operation)} Location.PhysicalLocation ${operation} "${value}")`;
       }
-    ],
+    ]
   ]);
 
-  private readonly queryTransformationOptions = new Map<AssetCalibrationFieldNames, TTLCache<string, unknown>>([
-    [AssetCalibrationFieldNames.LOCATION, this.systemAliasCache]
-  ]);
+  private multipleValuesQuery(field: AssetCalibrationFieldNames): ExpressionTransformFunction {
+    return (value: string, operation: string, _options?: any) => {
+      if (value.includes('{') && value.includes('}')) {
```

Comment:
I don't think this guarantees you the order.

For instance `a } = { b` would still work.

Reply:
Agree, added a function that will check for the placement of those chars as well.